### PR TITLE
Filters extraction function

### PR DESCRIPTION
**Pull request by gaetano-guerriero**

### CODE REVIEW

File: pydruid/utils/filters.py [hunk 1]
```diff
@@ -56,10 +60,25 @@ def __init__(self, **args):
...
                                       "alphaNumeric": args["alphaNumeric"]}}
+        elif args["type"] == "extraction":
+            self.filter = {"filter": {"type": "extraction",
+                                      "dimension": args["dimension"],
+                                      "value": args["value"]}}
         else:
             raise NotImplementedError(
                 'Filter type: {0} does not exist'.format(args['type']))
 
+        if extraction_function is not None:
```
Comment (se7entyse7en) [inline pydruid/utils/filters.py]: I'd move this at the beginning of the method.

File: pydruid/utils/filters.py [hunk 2]
```diff
@@ -20,13 +20,17 @@
 
 
 class Filter:
-    def __init__(self, **args):
 
-        if 'type' not in args.keys():
+    # filter types supporting extraction function
+    _FILTERS_WITH_EXTR_FN_ = ('selector', 'regex', 'javascript', 'in', 'bound',
```
Comment (se7entyse7en) [inline pydruid/utils/filters.py]: Way also the trailing underscore?

Comment (gaetano-guerriero) [inline pydruid/utils/filters.py]: hurry, fixed

File: pydruid/utils/filters.py (hunk 1 above)
Comment (gaetano-guerriero) [inline pydruid/utils/filters.py]: right, done

**Pull request merged**